flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMasterGateway.java

```diff
@@ -271,4 +275,19 @@ void heartbeatFromTaskManager(
 	 * @return The updated aggregate
 	 */
 	CompletableFuture<Object> updateGlobalAggregate(String aggregateName, Object aggregand, byte[] serializedAggregationFunction);
+
+	/**
+	 * Deliver a coordination request to a specified coordinator and return the response.
+	 *
+	 * @param operatorId identifying the coordinator to receive the request
+	 * @param serializedRequest serialized request to deliver
+	 * @return A future containing the response.
+	 *         The response will fail with a {@link org.apache.flink.util.FlinkException}
+	 *         if the task is not running, or no operator/coordinator exists for the given ID,
+	 *         or the coordinator cannot handle client events.
+	 */
+	CompletableFuture<CoordinationResponse> deliverCoordinationRequestToCoordinator(
```

Comment:
I think this should add to `JobMasterOperatorEventGateway`

Reply:
No. `JobMasterOperatorEventGateway` is for `OperatorEvent`s. `CoordinationRequest` and `CoordinationResponse` are different from `OperatorEvent`.